### DISCUSSION TITLE
Immich on a Raspberry Pi 4 - 8GB

### DISCUSSION THREAD

**Discussion by Eirikr70** [immich-app/immich#3807] [Q&A] [answered]
Hello you Great Sorcerers of Immich,
So I have been installing Immich this weekend for my 17,000 photos. It seems to be starting great. Anyway, I have a very high CPU usage (around 100% for more than 8h). All the pictures have been integrated more than 1 1/2 h ago and I still have CPU usage of machine learning container about 2.5-3 CPU out of 4. Since I have the dedicated fan, I am not nervous about it, if it slowly decreases and reaches a sustainable level. So my question is : will the CPU usage of machine learning remain that high so I have to renounce to it (and to Search functionality) ? Or will it slow down within a few hours/days so I can keep it on ?
Thanks a lot.

**Comment by mertalev** [comment, marked as answer]
It will take some time for it to complete with a Raspberry Pi 4, but once it does the load will be very low since it will only process the daily few images and videos you upload and idle outside of that.

**Reply by Eirikr70** [reply]
Thanks a lot for that reply. That is what I hoped. So I shall wait for it to complete.

**Reply by jrasm91** [reply]
You can check the status in admin > jobs to get a sense for how many are left and how long it is taking. You can also run machine learning temporarily on another machine.

**Reply by h4b4n3r0** [reply]
How would I run the CPU intense stuff on another machine?

**Reply by jrasm91** [reply]
https://immich.app/docs/guides/machine-learning

**Comment by Issomane** [comment]
For those running into the same issue, there is a way to limit the CPU usage of a process. 
With ```cpulimit``` you can for example set a limit to 50% of usage and it will leave 50% for other processes to run smoothly. 

- [cpulimit](https://manpages.ubuntu.com/manpages/trusty/man1/cpulimit.1.html) man page
- [Tutorial here](https://my.cloudfanatic.net/index.php/knowledgebase/27/How-to-Limit-CPU-Usage-of-a-Process-in-Linux-with-CPULimit.html)

TIP: Run `cpulimit` with the `-b` flag in order to run it in the background and unblock the use of your terminal
